src/workerd/jsg/setup.h

```diff
@@ -431,6 +434,15 @@ class Isolate: public IsolateBase {
     Isolate& jsgIsolate;
   };
 
+  void runInLockScope(auto func) {
```

Comment:
please, please add some comment about `func` signature! I've spent so much time deducing such things in our codebase...